src/main/java/net/backupcup/mcde/screen/RunicTableScreen.java

```diff
@@ -157,26 +161,27 @@ public void render(DrawContext ctx, int mouseX, int mouseY, float delta) {
             drawMouseoverTooltip(ctx, mouseX, mouseY);
             return;
         }
-        Enchantment enchantment = hoveredChoice.get().getEnchantment();
-        Identifier enchantmentId = hoveredChoice.get().getEnchantmentId();
+        var hovered = hoveredChoice.get();
+        Enchantment enchantment = hovered.getEnchantment();
+        Identifier enchantmentId = hovered.getEnchantmentId();
         List<Text> tooltipLines = new ArrayList<>();
         int level = 1;
         boolean enoughLevels = RunicTableScreenHandler.canEnchant(client.player, enchantmentId, level);
         MutableText enchantmentName = Text.translatable(enchantment.getTranslationKey())
             .formatted(EnchantmentUtils.formatEnchantment(enchantmentId));
-        if (hoveredChoice.get() instanceof ChoiceWithLevel withLevel) {
+        if (hovered.isChosen()) {
```

Comment:
This, in contrast to the previous `Chosen` comment, is a very well named method. It is referring to the action of the user and, as such, uses the `past participle` rather than a `noun` or `simple verb` for the method. Very good!